src/tools/mesh_tester.cpp

```diff
@@ -250,6 +328,109 @@ std::vector< std::pair<int, int> > naiveIntersectionAlgorithm(
   return retval;
 }
 
+#if defined(AXOM_USE_RAJA)
+template < typename ExecSpace >
+std::vector< std::pair<int, int> > naiveIntersectionAlgorithm(
+  mint::Mesh* surface_mesh,
+  std::vector<int> & degenerate)
+{
+  SLIC_INFO("Running naive intersection algorithm "
+    << " in execution Space: " 
+    << axom::execution_space< ExecSpace >::name());
+
+  // Get allocator
+  #ifdef AXOM_USE_UMPIRE
+    int allocatorID = axom::execution_space< ExecSpace >::allocatorID();
+    umpire::Allocator allocator = axom::getAllocator(allocatorID);
+  #endif
+
+  std::vector< std::pair<int, int> > retval;
+
+  const int ncells = surface_mesh->getNumberOfCells();
+  SLIC_INFO("Checking mesh with a total of "<< ncells<< " cells.");
+
+  #ifdef AXOM_USE_UMPIRE
+    Triangle3 * tris = axom::allocate <Triangle3> (ncells, allocator);
+  #else
+    Triangle3 * tris = axom::allocate <Triangle3> (ncells);
+  #endif
+
+  // Get each triangle in the mesh and check for degeneracies
+  for (int i = 0; i < ncells ; i++)
+  {
+    tris[i] = getMeshTriangle(i, surface_mesh);
+    if (tris[i].degenerate())
+    {
+      degenerate.push_back(i);
+    }
+  }
+
+  RAJA::RangeSegment row_range(0, ncells);
+  RAJA::RangeSegment col_range(0, ncells);
+
+  using KERNEL_POL = 
+    typename axom::mint::internal::structured_exec< ExecSpace >::loop2d_policy;
+  using REDUCE_POL =
+    typename axom::execution_space< ExecSpace >::reduce_policy;
+  using ATOMIC_POL = 
+    typename axom::execution_space< ExecSpace >::atomic_policy;
+ 
+  RAJA::ReduceSum< REDUCE_POL, int > numIntersect(0);  
+
+  // Compute the number of intersections
+  RAJA::kernel<KERNEL_POL>( RAJA::make_tuple(col_range, row_range),
+    AXOM_LAMBDA(int col, int row) {
+    if (row > col)
+    {
+      if (checkTT (tris[row], tris[col]))
+      {
+        numIntersect += 1;
+      }
+    }
+  });
+
+  // Allocation to hold intersection pairs and counter to know where to store
+  #ifdef AXOM_USE_UMPIRE
+    int * intersections =
+      axom::allocate <int> (numIntersect.get() * 2, allocator);
+    int * counter = axom::allocate <int> (1, allocator);
+  #else
+    int * intersections =
+      axom::allocate <int> (numIntersect.get() * 2);
+    int * counter = axom::allocate <int> (1);
```

Comment:
Would it be simpler to make `counter` a RAJA ReduceSum operation? 

Another, perhaps better alternative would be to use our `axom::StackArray` for this since the size is known a priori.

Reply:
Thanks @gzagaris.

``counter`` needs to be at a unique number across all threads since it allows each thread to add its data to a shared array, so I don't think a ``ReduceSum`` will work here.

What would be the benefit of a ``StackArray``, other than not needing to ``deallocate``?